projects/batfish/src/main/java/org/batfish/grammar/flatjuniper/ConfigurationBuilder.java

```diff
@@ -2246,10 +2256,27 @@ public void exitB_advertise_peer_as(B_advertise_peer_asContext ctx) {
     _currentBgpGroup.setAdvertisePeerAs(true);
   }
 
+  public void exitB_auth_algorithm(B_auth_algorithmContext ctx) {
+    if (ctx.AES_128_CMAC_96() != null) {
+      _currentBgpGroup.setAuthAlgorithm(BgpAuthenticationAlgorithm.AES_128_CMAC_96);
+    } else if (ctx.MD5() != null) {
+      _currentBgpGroup.setAuthAlgorithm(BgpAuthenticationAlgorithm.TCP_ENHANCED_MD5);
+    } else {
+      _currentBgpGroup.setAuthAlgorithm(BgpAuthenticationAlgorithm.HMAC_SHA_1_96);
+    }
+  }
+
+  public void exitB_auth_key(B_auth_keyContext ctx) {
+    _currentBgpGroup.setAuthKey(ctx.key.getText());
+  }
+
+  public void exitB_auth_key_chain(B_auth_key_chainContext ctx) {
+    _currentBgpGroup.setAuthKeyChainName(ctx.name.getText());
+  }
+
   @Override
   public void exitB_description(B_descriptionContext ctx) {
-    String description = ctx.description().text.getText();
```

Comment:
I don't have a strong opinion here, but why the change? Were you getting a warning?

Reply:
No, but I used inline assignment for other fields like tolerance, so I try to make it consistent.